src/Uno.UI/UI/Xaml/Controls/Repeater/ItemsRepeater.cs

```diff
@@ -0,0 +1,821 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Uno;
+using Uno.Disposables;
+using Uno.UI;
+using Uno.UI.Helpers.WinUI;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class ItemsRepeater : FrameworkElement, IPanel
+	{
+		private readonly UIElementCollection _repeaterChildren;
+		UIElementCollection IPanel.Children => _repeaterChildren;
+		internal IList<UIElement> Children => _repeaterChildren;
+
+
+		// Change to 'true' to turn on debugging outputs in Output window
+		//private static bool s_IsDebugOutputEnabled = false;
+
+		internal static Point ClearedElementsArrangePosition = new Point(-10000.0f, -10000.0f);
+
+		// A convention we use in the ItemsRepeater codebase for an invalid Rect value.
+		internal static Rect InvalidRect = new Rect(-1, -1, -1, -1);
+
+
+
+		public event TypedEventHandler<ItemsRepeater, ItemsRepeaterElementPreparedEventArgs> ElementPrepared;
+		public event TypedEventHandler<ItemsRepeater, ItemsRepeaterElementIndexChangedEventArgs> ElementIndexChanged;
+		public event TypedEventHandler<ItemsRepeater, ItemsRepeaterElementClearingEventArgs> ElementClearing;
+
+		// Cached Event args to avoid creation cost every time
+		private ItemsRepeaterElementPreparedEventArgs m_elementPreparedArgs;
+		private ItemsRepeaterElementClearingEventArgs m_elementClearingArgs;
+		private ItemsRepeaterElementIndexChangedEventArgs m_elementIndexChangedArgs;
+
+		//public Microsoft.UI.Xaml.Controls.IElementFactoryShim ItemTemplateShim() { return m_itemTemplateWrapper; };
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)